### ISSUE TITLE
Support associations to reduce calls to server

### ISSUE THREAD

**Issue by sthaha** [product-definition-center/pdc-ruby-gem#6]
The following code
```ruby
release = Release.find('fedora-22')
release.variants.map(&:release) 
```
Fetches release X times instead of 0. Perhaps the caching may stop this from happening but it may be worth fixing by adding support of association.

**Comment by ycheng-aa** [comment]
@sthaha Does it mean we should port Spyke association implementation to PDC gem?

**Comment by sthaha** [comment]
@ycheng-aa yes, most of the gem is based on spyke which already has associations. We could base our implementation on spyke.
